File: aoc-2022/day-06/python/part2_sol.py
f = open("../input.txt")
data = f.read().rstrip()


# data = "mjqjpqmgbljsphdztnvjfqwrcgsmlb"

for i in range(13, len(data)):
    s = set()
    for c in data[i-13:i+1]:
        s.add(c)
    if len(s) == 14:
        print(i+1)
        break

File: 2022/day-06/rust/part2_sol.rs
use std::fs;
use std::collections::HashSet;

fn main() {
    let data = fs::read_to_string("../input.txt").expect("Couldn't read from file");
    for i in 14..data.len() {
        let mut set = HashSet::new();
        for c in data[i-14..i].chars() {
            set.insert(c);
        }
        if set.len() == 14 {
            println!("{}", i);
            break;
        }
    }
}